app/controllers/checkouts_controller.rb

```diff
@@ -3,13 +3,26 @@ class CheckoutsController < ApplicationController
 
   def create
     @book = Book.find(params[:book_id])
-    checkout = Checkout.create(book: @book, user: current_user)
+    if CheckOutBook.process(book: @book, user: current_user)
+      flash[:success] = t('.successful_message')
+      redirect_to root_path
+    else
+      flash[:error] = t('.unsuccessful_message')
```

Comment:
You're not redirecting in this branch though. So I think if you did get an error in the checkout process somehow (maybe someone else checked the book out right before you did), I think rails would raise an error because it will attempt to render the :create view by default. Same with the update action below, actually.

I think you need the redirect_to call here too.